src/Uno.UI.Runtime.Skia.Android/AndroidSkiaTextBoxNotificationsProviderSingleton.cs

```diff
@@ -31,7 +32,15 @@ public void OnUnfocused(TextBox textBox)
 	{
 		if (UnoSKCanvasView.Instance is { } canvasView)
 		{
-			canvasView.TextInputPlugin.HideTextInput();
+			// Hide the keyboard only when the next focused element is not a TextBox
+			// This prevents the keyboard from flickering when switching between TextBoxes
+			// https://github.com/unoplatform/uno-private/issues/1160
+			var xamlRoot = textBox.XamlRoot;
+			if (xamlRoot is null || FocusManager.GetFocusedElement(xamlRoot) is not TextBox)
```

Comment:
in the case of ComboBox(IsEditable=true) or NumberBox getting focus, will GetFocusedElement return its inner textbox or the itself? if the later, this condition could fail

Reply:
This comment came after the Pr was merged, I will push a second Pr covering this comment